### PR TITLE
refactor(dateHelper): swatch-260 remove moment.js from dateHelper

### PR DESCRIPTION
**Pull request by vbusch**
## What's included
[sw-260](https://issues.redhat.com/browse/SWATCH-260) Replace moment.js

### Notes
Removed moment.js from the date helper.  This is a non functional change.  There should be no noticeable UI differences.
TODO: remove it from the other files.

### Coverage and basic unit test check
1. update the NPM packages with $ yarn
2. $ yarn test
3. confirm tests come back clean

### Proxy run check
1. update the NPM packages with $ yarn
2. make sure Docker is running, plus on network, then
3. $ yarn start:proxy
4. http://localhost:3000/apps/subscriptions/rhel
5. change the time range for the cpu socket usage.  And verify the network request has the correct range.

### Check the build
1. update the NPM packages with $ yarn
2. $ yarn build
3. confirm tests come back clean

## Updates issue/story
[sw-260](https://issues.redhat.com/browse/SWATCH-260)

### CODE REVIEW

**Comment by mirekdlugosz** [comment]
@vbusch if you have access to `iqe-curiosity-plugin` on internal GitLab, please take a look at `iqe_curiosity/tests/unittests_utils.py`. There's a number of unit tests that cover cases that I have seen failing over the years. I think it might be good idea to add at least some of them to unit tests here, to ensure this change is backwards compatible.

@diegomaranhao can help you with accessing these tests, and is up to date with whether they are still running or not.

File: src/common/__tests__/dateHelpers.test.js
Comment (cdcabrera) [inline src/common/__tests__/dateHelpers.test.js:57]: consistency, lets update the `it should` copy to be consistent for all of the new tests
```
it('should setEndOfDay date to...
it('should setStartOfDay date to...
etc 
```

File: src/common/__tests__/dateHelpers.test.js
Comment (cdcabrera) [inline src/common/__tests__/dateHelpers.test.js:78]: add in the test for the default option changes noted above

optional
consider merging this with the existing `setRangedDateTime` test around line `32`

**Review by cdcabrera — CHANGES_REQUESTED** [review]
All minor stuff, mostly copy. Looks good at the functional level!

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js]: Updated

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js:27]: descriptions are now added.

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js]: Modified.

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js]: default added with coverage.

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js]: I added the annotations.  But we can do another PR to make them private.

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js]: done

File: src/common/__tests__/dateHelpers.test.js
Comment (vbusch) [reply src/common/__tests__/dateHelpers.test.js]: updated so they all start with 'should...'

File: src/common/__tests__/dateHelpers.test.js
Comment (vbusch) [reply src/common/__tests__/dateHelpers.test.js]: merged and added.

File: src/common/dateHelpers.js
Comment (cdcabrera) [inline src/common/dateHelpers.js:57]: this was something that was overlooked in the original and since we don't really run `setRangedDateTime` as `setRangedDateTime()` anywhere this doesn't really show up readily...  adding this fix here... 
```
... days' } = {})
```
lets the function run standalone with the parameter defaults...

(if you add a unit test for running `setRangedDateTime()` you'll see it)

**Review by cdcabrera — CHANGES_REQUESTED** [review]
minor parameter adjustment

File: src/common/dateHelpers.js
Comment (vbusch) [reply src/common/dateHelpers.js]: fixed.

**Review by cdcabrera — COMMENTED** [review]
lgtm!

stepped through the combinations against data in staging, ranges appear to maintain the original ranges for standard granularity (daily, weekly, monthly, quarterly), and the additional "ranged granularity" we use for on-demand displays

@vbusch this is an approval without being a github approval, reason, before we merge this update, reviewing a sync between envs for variant work

**Review by cdcabrera — APPROVED** [review]

**Pull request merged**
